modules/ztp-deploying-a-site.adoc

```diff
@@ -0,0 +1,187 @@
+      group-du-sno: ""
+      # ../policygentemplates/example-sno-site.yaml will apply to all clusters with 'sites: "example-sno"'
+      # Normally this should match or contain the cluster name so it only applies to a single cluster
+      sites : "example-sno"
+    clusterNetwork:
+      - cidr: 1001:1::/48
+        hostPrefix: 64
+    machineNetwork: <3>
+      - cidr: 1111:2222:3333:4444::/64
+      # For 3-node and standard clusters with static IPs, the API and Ingress IPs must be configured here
+    apiVIP: 1111:2222:3333:4444::1:1 <4>
+    ingressVIP: 1111:2222:3333:4444::1:2 <5>
+
+    serviceNetwork:
+      - 1001:2::/112
+    additionalNTPSources:
+      - 1111:2222:3333:4444::2
+    nodes:
+      - hostName: "example-node1.example.com" <6>
+        role: "master"
+        bmcAddress: "idrac-virtualmedia+https://[1111:2222:3333:4444::bbbb:1]/redfish/v1/Systems/System.Embedded.1" <7>
+        bmcCredentialsName:
+          name: "example-node1-bmh-secret" <8>
+        bootMACAddress: "AA:BB:CC:DD:EE:11"
+        bootMode: "UEFI"
+        rootDeviceHints:
+          hctl: '0:1:0'
+        cpuset: "0-1,52-53"
+        nodeNetwork: <9>
+          interfaces:
+            - name: eno1
+              macAddress: "AA:BB:CC:DD:EE:11"
+          config:
+            interfaces:
+              - name: eno1
+                type: ethernet
+                state: up
+                macAddress: "AA:BB:CC:DD:EE:11"
+                ipv4:
+                  enabled: false
+                ipv6:
+                  enabled: true
+                  address:
+                  - ip: 1111:2222:3333:4444::1:1
+                    prefix-length: 64
+            dns-resolver:
+              config:
+                search:
+                - example.com
+                server:
+                - 1111:2222:3333:4444::2
+            routes:
+              config:
+              - destination: ::/0
+                next-hop-interface: eno1
+                next-hop-address: 1111:2222:3333:4444::1
+                table-id: 254
+----
+<1> Applies to all cluster types. The value must match an image set available on the hub cluster. To see the list of supported versions on your hub, run `oc get clusterimagesets`.
+<2> Applies to all cluster types. These values must correspond to the `PolicyGenTemplate` labels that you define in a later step.
+<3> Applies to single node clusters. The value defines the cluster network sections for a single node deployment.
+<4> Applies to three-node and standard clusters. The value defines the cluster network sections.
+<5> Applies to three-node and standard clusters. The value defines the cluster network sections.
+<6> Applies to all cluster types. For single node deployments, define one host. For three-node deployments, define three hosts. For standard deployments, define three hosts with `role: master` and two or more hosts defined with `role: worker`.
```

Comment:
I think in here we'd need to use the same terminology as Openshift uses. I wonder if this applies: https://access.redhat.com/webassets/avalon/d/OpenShift_Container_Platform-4.10-Architecture-en-US/images/9fb14e59171a618c486862da6b1ae17b/create-nodes.png (taken from https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/architecture/architecture-installation#installation-process_architecture-installation)
and the usage of "controller" and "compute" instead of "master" and "worker".

Reply:
This may impact the code. @imiller0 - can these be changed?

Reply:
I believe that these roles are hard coded in the Assisted Service and so we are unfortunately constrained to these terms.
https://github.com/openshift/assisted-service/blob/66ee39b1c1192e35dce7f9d4b3aba1c50336ca85/models/host_role.go#L33-L36